server/src/main/java/org/elasticsearch/action/ingest/PutPipelineRequest.java

```diff
@@ -25,13 +25,15 @@
 import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentHelper;
 import org.elasticsearch.common.xcontent.XContentType;
 
 import java.io.IOException;
 import java.util.Objects;
 
-public class PutPipelineRequest extends AcknowledgedRequest<PutPipelineRequest> {
+public class PutPipelineRequest extends AcknowledgedRequest<PutPipelineRequest> implements ToXContentObject {
```

Comment:
I'm not entirely sure if we want to add ToXContent here. Other requests where the body consist of a BytesReference source (e.g.  IndexRequest) don't seem to do this, instead we convert them to a ByteArrayEntity in RequestConverters. Not sure whats the benefit of either of those choices, maybe @javanna has an opinion on this.

Reply:
I see, I was curious. I don't have a strong opinion either way at the moment, interested in hearing the pros and cons that others might think of

Reply:
I have no strong opinion either, it can be convenient to implement ToXContentObject I think. IndexRequest is a bit of a different one as the body is the document itself.